doc/appendices/administration.rst

```diff
@@ -150,5 +150,12 @@ After committing the release tag:
   (either manually or using ``spack checksum``) and submit a pull request to
   the Spack project.
 
+The first commit that deviates from the most recent major or minor branch
+should be tagged (but not released!) with the next version number with a
+``-dev`` suffix. For example, after releasing version 1.0.0, the next
+commit on the main branch that is *not* intended for version 1.0.1 (i.e., the
```

Comment:
Why not simplify a bit and do so on the first commit?  Likely there will be other commit afterward that would be needing to be applied as patch on top the release, why would the first few one be "special"?

Reply:
if you look at the graph between v0.1.2 and the current "master" (which is the 0.2 dev) it only makes sense to mark the point at which we diverged from bugfixes.
```console
$ git log --cherry-mark --graph --oneline upstream/master upstream/backports/v0.1 v0.1.0
* 16c5d530 (upstream/backports/v0.1, backports/v0.1) Reset physics state when a sec>
* <snip>
* 14984701 (tag: v0.1.3) Add release notes and update documentation
* <snip>
* 41e58e1e (tag: v0.1.2) Add release docs for v0.1.2
* <snip>
| * 904987c5 (upstream/master, upstream/HEAD) Reset physics state when a secondary >
| * <snip>
| * d2c76ea5 (tag: v0.2.0-dev) Add granularity to Geant setup (#485)
|/
* 531dd911 (tag: v0.1.1) Add contributing guide and development docs (#502)
* <snip>
* 04fe945d (tag: v0.1.0) Add release notes and final doc tweaks for v0.1.0 (#468)
```

Reply:
Fair enough.  It works for now.  In the (much) later stage of development the branch may actually happen before the actual tagging.  For example, we may propose to the users a release candidate and based on their feedback apply fixes while in parallel new development has happened.  (But I guess the text is then still correct just essentially replacing "release" with "release candidate").